integr8ly/integreatly-operator
# MGDAPI-3202 - RHOAM Multitenant on demand (crd based solution)

# Issue link
https://issues.redhat.com/browse/MGDAPI-3202
Make RHOAM Multitenant on demand

# What
- We want to make rhoam multitenant tenants created based on the presence of new CR of type RhoamTenant, 
where CR name to be same as username. Based on this user will be annotated as "tenant"
- RHMI controller will watch for change in user (no changes in rhmi controller)
- 3scale reconciler will get tenant users by annotation 
( userHelper.GetMultiTenantUsers() method changed to get tenant users by annotation instead of timestamp)

# Verification steps
1. Prepare Sandbox Rhoam cluster
1.1. Check that RhoamTenant CRD created after `make cluster/prepare/local`
1.2. Create IDP, and have an any user for testing (like customer-admin01, or test, any ...). Check that user have no "tenant" annotation before test
1.3. Create Instance of RhoamTenant crd. CR should have same names as user, like "customer-admin01".
1.4 Check that your test user annotated, as `Annotations:	tenant=yes`
1.5.Create one more CR, with name as non existing user. Save CR. Check CR status. You will see something like following: Last Error: error get user with name as current CR, users.user.openshift.io "customer1" not found

2. Prepare NON Sandbox Rhoam cluster
and repeat same steps. Nothing should happen, users will not be annotated, even if CR exists
